unix/tar_unix.ml

```diff
@@ -79,11 +79,15 @@ let write_end = T.write_end
 module Archive = struct
   include T.Archive
 
+  let with_file name flags perms f =
+    let fd = Unix.openfile name flags perms in
+    try f fd with e -> Unix.close fd; raise e
```

Comment:
Shouldn't this close it in the success case too?

Reply:
Of course, what was I thinking.
Is `Fun.protect` suited for the job here?

Reply:
Should do, yes.